src/App.tsx

```diff
@@ -1,27 +1,95 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import classNames from 'classnames';
+import { GoodsList } from './components/GoodsList';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { LoadingError } from './components/LoadingError';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoad] = useState({
+    all: false,
+    five: false,
+    red: false,
+  });
+  const [loadingError, setloadingError] = useState(false);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleGoods = async (type: string) => {
```

Comment:
You may leave it as it is now, but this solution is not scalable
In case if you pass loader function as parameter to this function - you can create as many functions as you want, but now for that you need to add another case in this function to handle it